packages/client/lib/miner/pendingBlock.ts

```diff
@@ -163,33 +201,40 @@ export class PendingBlock {
       )
       this.constructBlobsBundle(payloadId, blobTxs, header.hash())
     }
-    return payloadId
+    return payloadIdBuffer
   }
 
   /**
    * Stops a pending payload
    */
-  stop(payloadId: Buffer) {
-    const payload = this.pendingPayloads.find((p) => p[0].equals(payloadId))
-    if (!payload) return
+  stop(payloadIdBuffer: Buffer | string) {
+    const payloadId =
+      typeof payloadIdBuffer !== 'string' ? bufferToHex(payloadIdBuffer) : payloadIdBuffer
+    const builder = this.pendingPayloads.get(payloadId)
+    if (!builder) return
```

Comment:
In other code (which is not edited here but is in the same files) we explicitly check `builder !== undefined`, should we change? (This is also part of an eslint discussion we had a while back to make these checks more strict)